api/src/main/java/io/minio/HttpRequestBody.java

```diff
@@ -71,7 +71,7 @@ public long contentLength() {
   public void writeTo(BufferedSink sink) throws IOException {
     if (data instanceof InputStream) {
       InputStream stream = (InputStream) data;
-      sink.write(Okio.source(stream), len);
+      sink.writeAll(Okio.source(stream));
```

Comment:
Using `sink.WriteAll()` is incorrect i.e. all data of `stream` is written, only `len` needs to be written.

We would need to find root cause of `sink.write()` failure.

Reply:
This is acceptable for now @balamurugana - root cause for write() is perhaps in the HTTP 2.0 implementation of reading RequestBody... I haven't found the root cause yet.

But this fixes everything for now. I suggest we take this in and move forward.

Reply:
Reading from input stream beyond given size leads to data loss (for user).  e.g. [putObject()](http://minio.github.io/minio-java/io/minio/MinioClient.html#putObject-java.lang.String-java.lang.String-java.io.InputStream-long-java.util.Map-) can use partial stream upload and rest of data for other purposes.

I am OK for workarounds.  Unfortunately we don't know the problem yet to solve here.

Reply:
I don't know why you think it's data loss, content is set and server honors it. If more data is sent server fails. So either way it's a moot argument. What we are trying to test is does write return an exception which is marginal..